cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramNotificationSubscriberService.java

```diff
@@ -415,13 +415,25 @@ private void handleProgramEvent(ProgramRunId programRunId, ProgramRunStatus prog
         recordedRunRecord = appMetadataStore.recordProgramStopping(programRunId, messageIdBytes, stoppingTsSecs,
                                                                    terminateTsSecs);
         writeToHeartBeatTable(recordedRunRecord, stoppingTsSecs, programHeartbeatTable);
+        if (logicalStartTimeSecs == -1) {
+          LOG.warn("Ignore program running notification for program {} without {} specified, {}",
+                  programRunId, ProgramOptionConstants.LOGICAL_START_TIME, notification);
+          return;
+        }
+        emitStartingTimeMetric(programRunId, logicalStartTimeSecs, programRunStatus, recordedRunRecord);
```

Comment:
Why do we need to emit the metric here? `SUSPENDED` is also not an `END_STATE` according to https://github.com/cdapio/cdap/blob/2f8e00aef38936104eef788fed7e7e0a11fe070b/cdap-proto/src/main/java/io/cdap/cdap/proto/ProgramRunStatus.java#L43